homeassistant/components/peco/sensor.py

```diff
@@ -34,18 +33,26 @@ async def async_setup_entry(
 
     async def async_update_data():
         """Fetch data from API."""
-        try:
-            async with async_timeout.timeout(10):
-                if conf["county"] == "TOTAL":
-                    return await api.get_outage_totals(websession)
-                else:
-                    return await api.get_outage_count(conf["county"], websession)
-        except InvalidCountyError as err:
-            raise ConfigEntryAuthFailed from err
-        except HttpError as err:
-            raise UpdateFailed from err
-        except BadJSONError as err:
-            raise UpdateFailed from err
+        async with async_timeout.timeout(10):
```

Comment:
The aiohttp ClientSession already has a timeout (I don't know how long). You can just except for `asyncio.TimeoutError` here.